### PR TITLE
Python 3 Support

### PR DESCRIPTION
**Pull request by seanwiseman**
As `elife-article` is a dependency of `elife-bot` we need to bring in Python 3 support in order to allow full Python 3 support for `elife-bot`.

I have made the initial changes. There is one test that was failing from the start for both Python versions:

`test_article.py` line 200. (I have added comments in the file)

The remaining 5 failing tests for Python 3 are all in `test_parse_deep.py` and are to do with comparing Unicode string values. I wanted to let you see these @gnott  as I did not want to modify the actual test code behaviour. You will see the values in the diff when running the tests.

### CODE REVIEW

File: elifearticle/article.py [hunk 1]
```diff
@@ -22,6 +26,23 @@ def __unicode__(self):
...
+        """
+        Return `str` representation of the simple object properties,
+        if there is a list or dict just return an empty representation
+        for easier viewing and test case scenario writing
+        """
+        _dict = {}
+        for key in self.__dict__:
+            if isinstance(self.__dict__.get(key), list):
+                _dict[key] = []
+            elif isinstance(self.__dict__.get(key), dict):
```
Comment (giorgiosironi) [inline elifearticle/article.py]: is `OrderedDict` used around here and/or falling into this conditional?

File: tests/test_parse_deep.py [hunk 2]
```diff
@@ -14,6 +17,8 @@ class TestParseDeep(unittest.TestCase):
     def test_parse_article_02935_simple(self):
         "some simple comparisons and count list items"
         article_object, error_count = parse.build_article_from_xml(XLS_PATH + 'elife-02935-v2.xml')
+        # test pretty method for test coverage
+        self.assertIsNotNone(article_object.pretty())
```
Comment (giorgiosironi) [inline tests/test_parse_deep.py]: could be stricter and test the return value is a string?

Comment (seanwiseman) [reply tests/test_parse_deep.py]: Upon checking it actually returns an `OrderedDict`:
```
def pretty(self):
    "sort values and format output for viewing and comparing in test scenarios"
    pretty_obj = OrderedDict()
    for key, value in sorted(iteritems(self.__dict__)):
        if value is None:
            pretty_obj[key] = None
        elif is_str_or_unicode(value):
            pretty_obj[key] = self.__dict__.get(key)
        elif isinstance(value, list):
            pretty_obj[key] = []
        elif isinstance(value, dict):
            pretty_obj[key] = {}
        else:
            pretty_obj[key] = unicode_value(value)
    return pretty_obj
```
I avoided changing the actual test assertions, I could change this one obviously its only a small change but even just in that test file there is about 30+ other cases that could check return type but just in fact make sure the return type is not `None`.

I will leave for now, but agree could be picked up during a refactor.

File: elifearticle/article.py (hunk 1 above)
Comment (seanwiseman) [reply elifearticle/article.py]: No its not and I think its right to not try and imply order at this level, the issue seemed to stem even from retrieving nested values (old code thats now been fixed):
`self.assertEqual(unicode_value(article_object.dates.get('pub')), "{'date_type': u'pub', ....some values}`

**Comment by seanwiseman** [comment]
@gnott I think I have checked all I can now. Is there anything else you would like to test or add?

File: elifearticle/article.py (hunk 1 above)
Comment (gnott) [reply elifearticle/article.py]: There were some fixture file comparisons in tests before, but due to the problems with converting to unicode I deleted them. I only had them there to speed up development until I was able to create individual tests and then the full file comparisons were no longer required.

The string values of object I would hope would only be used in tests, and the Article ``pretty()`` function has no real purpose (I think) unless someone thinks of one in the future.

File: tests/test_parse_deep.py (hunk 2 above)
Comment (gnott) [reply tests/test_parse_deep.py]: As above, the test is really only there for test coverage. If no one can think of a use case for ``pretty()`` then we can delete the function entirely.

**Comment by gnott** [comment]
I like it, thanks for all your efforts @seanwiseman and @giorgiosironi !

**Pull request merged**